ansible/roles/test/files/ptftests/py3/dhcp_relay_test.py

```diff
@@ -240,8 +245,75 @@ def create_dhcp_discover_relayed_packet(self):
+        ip_server="0.1.2.3",
+        ip_dst="255.255.255.255",
+        ip_offered="8.9.10.11",
+        port_dst=DHCP_CLIENT_PORT,
+        netmask_client="255.255.255.0",
+        ip_gateway=DEFAULT_ROUTE_IP,
+        dhcp_lease=256,
+        padding_bytes=0,
+        set_broadcast_bit=False,
+    ):
+        """
+        Return a DHCPOFFER packet
+        Supports a few parameters:
+        @param eth_server MAC address of DHCP server
+        @param eth_dst MAC address of destination (DHCP Client, Relay agent) or broadcast (ff:ff:ff:ff:ff:ff)
+        @param eth_client MAC address of DHCP client
+        @param ip_server IP address of DHCP server
+        @param ip_dst IP address of destination (DHCP Client, Relay agent) or broadcast (255.255.255.255)
+        @param ip_offered IP address that server is assigning to client
+        @param ip_gateway Gateway IP Address, address of relay agent if encountered
+        @param port_dst Destination port of packet (default: DHCP_PORT_CLIENT)
+        @param netmask_client Subnet mask of client
+        @param dhcp_lease Time in seconds of DHCP lease
+        @param padding_bytes Number of '\x00' bytes to append to end of packet
+        Destination IP can be unicast or broadcast (255.255.255.255)
+        Source port is always 67 (DHCP server port)
+        Destination port by default is 68 (DHCP client port), but can be also be 67 (DHCP server port) if being sent to a DHCP relay agent
+        """
+        my_chaddr = binascii.unhexlify(eth_client.replace(':', ''))
+        my_chaddr += b'\x00\x00\x00\x00\x00\x00'
+
+        pkt = scapy.Ether(dst=eth_dst, src=eth_server, type=self.DHCP_ETHER_TYPE_IP)
+        pkt /= scapy.IP(src=ip_server, dst=ip_dst, ttl=128, id=0)
+        pkt /= scapy.UDP(sport=self.DHCP_SERVER_PORT, dport=port_dst)
+        pkt /= scapy.BOOTP(
+            op=self.DHCP_BOOTP_OP_REPLY,
+            htype=self.DHCP_BOOTP_HTYPE_ETHERNET,
+            hlen=self.DHCP_BOOTP_HLEN_ETHERNET,
+            hops=0,
+            xid=0,
+            secs=0,
+            flags=self.DHCP_BOOTP_FLAGS_BROADCAST_REPLY if set_broadcast_bit else 0,
+            ciaddr=self.DEFAULT_ROUTE_IP,
+            yiaddr=ip_offered,
+            siaddr=ip_server,
+            giaddr=ip_gateway,
+            chaddr=my_chaddr,
+        )
+        pkt /= scapy.DHCP(
+            options=[
+                ("message-type", "offer"),
+                ("server_id", ip_server),
+                ("lease_time", int(dhcp_lease)),
+                ("subnet_mask", netmask_client),
+                (82, self.option82),
+                ("vendor_class_id", "http://10.23.145.67:9876/abc/defghijk/lmnopqrst.bin".encode('utf-8')),
+                ("end"),
+            ]
+        )
+        if padding_bytes:
+            pkt /= scapy.PADDING("\x00" * padding_bytes)
+        return pkt
+
     def create_dhcp_offer_packet(self):
-        return testutils.dhcp_offer_packet(eth_server=self.server_iface_mac,
```

Comment:
>testutils.dhcp_offer_packet [](http://example.com/codeflow?start=15&length=27)

This function is also used in tests/dhcp_relay/test_dhcp_pkt_fwd.py. Do you need to change that file?

Reply:
I replace this function to add option82, and I don't think we need it for test_dhcp_pkt_fwd.